src/components/filters/AttributeFilter/AttributeElements.tsx

```diff
@@ -198,16 +200,25 @@ export class AttributeElements extends React.PureComponent<IAttributeElementsPro
                     );
         });
 
+        let currentGetValidElementsPromise: Promise<any> = null;
+        // The promise needs to reset here because we are not setting it synchroneously
+        // and even this small delay is sometimes too late
+        this.getValidElementsPromise = null;
+
         uriPromise
             .then((uri: string) => {
                 const objectId = getObjectIdFromUri(uri);
-                return this.sdk.md.getValidElements(
+                this.getValidElementsPromise = currentGetValidElementsPromise = this.sdk.md.getValidElements(
```

Comment:
I know it's a shorthand, but I'd still assign it in multiple rows, as it's more clear to read